doc/ant-xgboost_user_guide.md

```diff
@@ -0,0 +1,265 @@
+
+<br>
+
+#### Quick Start
+
+* train XGBoost model on [boston housing dataset](https://www.kaggle.com/c/boston-housing)
+
+```sql
+SELECT *
+FROM boston.train
+TRAIN xgboost.Regressor
+WITH
+    train.num_round = 30,
+    train.eta = 0.15,
+    train.max_depth = 5,
+    train.subsample = 0.9
+COLUMN crim, zn, indus, chas, nox, rm, age, dis, rad, tax, ptratio, b, lstat
+LABEL medv
+INTO sqlflow_models.my_boston_xgb_model;
+```
+
+* predict with trained XGBoost model
+
+```sql
+SELECT *
+FROM boston.test
+PREDICT boston.prediction.result
+WITH
+    pred.append_columns = [crim, zn, indus, chas, nox, rm, age, dis, rad, tax, ptratio, b, lstat, medv]
+USING sqlflow_models.my_boston_xgb_model;
+```
+
+* show some prediction results
+
+```sql
+SELECT * FROM boston.prediction LIMIT 10;
+```
+
+* try auto train 
+
+```sql
+SELECT *
+FROM boston.train
+TRAIN xgboost.Regressor
+WITH
+    train.auto_train = true
+COLUMN crim, zn, indus, chas, nox, rm, age, dis, rad, tax, ptratio, b, lstat
+LABEL medv
+INTO sqlflow_models.my_boston_autoxgb_model;
+```
+
+<br>
+
+#### Overall SQL Syntax
+##### Training Syntax
+```sql
+// standard select clause
+SELECT ... FROM ${TABLE_NAME}
+// train clause
+TRAIN xgboost.${estimatorType}
+WITH
+    [optional] ${train.attributes}
+    ......
+    ......
+COLUMN ${feature_columns}
+[optional] COLUMN ${group_column} FOR group
+[optional] COLUMN ${weight_column} FOR weight
+LABEL ${label_column}
+INTO ${model};
+```
+##### Prediction Syntax
+```sql
+// standard select clause
+SELECT ... FROM ${TABLE_NAME}
+// pred clause
+PREDICT ${output_table}.${result_column}
+WITH
+    [optional] ${pred.attributes}
+    ......
+USING ${model};
+```
+<br>
+
+#### Estimators
+We provide various XGBoost estimators for better user experience.
+All of them are case-insensitive and sharing same prefix `xgboost`. They are listed below.
+
+* xgboost.Estimator
+
+  General estimator, with this `train.objective` should be defined explicitly.
+  
+* xgboost.Classifier 
+    
+  Estimator for classification task, works with `train.num_class`. Default value is binary classification.
+  
+* xgboost.BinaryClassifier 
+    
+  Estimator for binary classification task, set `train.objective` to `binary:logistic`.
+
+* xgboost.MultiClassifier 
+    
+  Estimator for multi classification task, set `train.objective` to `multi:softprob`, should work with `train.num_class` > 2.
+
+* xgboost.Regressor 
+    
+  Estimator for regression task, set `train.objective` to `reg:squarederror`(`reg:linear`). 
+
+<br>
+
+#### Columns
+
+##### Feature Columns
+For now, two feature column schemas are available.
+
+First one is `dense schema`, which concatenate numeric table columns transparently, such as `COLUMN f1, f2, f3, f4`.
+
+Second one is `sparse key-value schema`, which received string sparse feature formatted like `$k1:$v1,$k2:$v2,...`.
+This schema is decorated with keyword `SPARSE`, such as `COLUMN SPARSE(col1)`.
```

Comment:
Is `SPARSE` currently supported?

Reply:
Yes, but we need some tests for `SPARSE`.